api/api/utils/photon.py

```diff
@@ -99,6 +139,19 @@ def get(
         raise UpstreamThumbnailException(
             f"Failed to render thumbnail due to timeout: {exc}"
         )
+    except requests.HTTPError as exc:
+        # Save occurrences to redis and only sent to Sentry when reached a
+        # threshold to prevent providers downtime consuming all the events
+        # quota.
+        key = f"{settings.THUMBNAIL_HTTP_ERROR_PREFIX}{domain}"
+        errors = 1
+        try:
+            errors = cache.incr(key)
+        except ValueError:  # Key does not exist.
+            cache.set(key, 1)
+        if errors >= settings.THUMBNAIL_HTTP_ERROR_THRESHOLD_TO_NOTIFY:
+            sentry_sdk.capture_exception(exc)
+        raise UpstreamThumbnailException(f"Failed to render thumbnail. {exc}")
```

Comment:
Great idea planning for sentry throttling in this manner! Do you think it would make sense to add a test to simulate this case and assert that `sentry_sdk.capture_exception` isn't called until the threshold is hit?

Reply:
Added. Thanks to @sarayourfriend that made all the setup a breeze to built this test ⭐️